### PR TITLE
Add LLM2CLIP (OpenAI variants)

### LINKED ISSUE #1458: [mieb] add support for LLM2CLIP
https://github.com/microsoft/LLM2CLIP

I can take look at this

### PR DESCRIPTION
**Pull request by isaac-chung**
- Add LLM2CLIP (OpenAI variants)
- Fixes #1458 
- For `microsoft/LLM2CLIP-Openai-L-14-224`
  - Needed to edit [modeling_clip.py](https://huggingface.co/microsoft/LLM2CLIP-Openai-L-14-224/discussions/1/files#d2h-634177) locally to use relative import.

### Usage
```
mteb run -t CIFAR10ZeroShot -m microsoft/LLM2CLIP-Openai-L-14-336
```

### Code Quality

- [x] **Code Formatted**: Format the code using `make lint` to maintain consistent style.

### Documentation

- [x] **Updated Documentation**: Add or update documentation to reflect the changes introduced in this PR.

### Testing

- [x] **New Tests Added**: Write tests to cover new functionality. Validate with `make test-with-coverage`.
- [x] **Tests Passed**: Run tests locally using `make test` or `make test-with-coverage` to ensure no existing functionality is broken.

### CODE REVIEW

File: mteb/models/llm2clip_models.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+import torch
+from PIL import Image
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+from transformers import AutoConfig, AutoModel, AutoTokenizer, CLIPImageProcessor
+
+from mteb.encoder_interface import PromptType
+from mteb.model_meta import ModelMeta
+
+MODEL2PROCESSOR = {
```
Comment (Samoed) [inline mteb/models/llm2clip_models.py:16]: Maybe it would be better to pass directly processors to the loader like in RepLLama?
https://github.com/embeddings-benchmark/mteb/blob/2dd13912b6e5c7c8be93aeb9f7dd873671309961/mteb/models/repllama_models.py#L131-L135

**Comment by isaac-chung** [comment]
Ran on CIFAR10ZeroShot:
| Model Name  | Accuracy |
|------------|-------------|
| B-16       | 90.56       |
| L-14-224   | 94.80       |
| L-14-336   | 94.33       |

**Review by gowitheflow-1998 — APPROVED** [review]
